File: dojbot/server/geminiIntegration.js
// Import Google Generative AI SDK
const { GoogleGenerativeAI } = require("@google/generative-ai");

// Set up API Key (use environment variable for security)
const genAI = new GoogleGenerativeAI("AIzaSyAi4RcPWCz6gP2dff23o65fXoi4a_9lXek");
const model = genAI.getGenerativeModel({ model: "gemini-1.5-flash" });

// Function to send the question to Gemini API and return the response
const generateResponse = async (question) => {
  try {
    const result = await model.generateContent(question);
    
    // Extract the response text (from candidates array)
    const responseText = result.candidates[0].output || "No response generated.";
    
    return responseText;
  } catch (error) {
    console.error("Error calling Gemini API:", error);
    throw new Error("Error generating content.");
  }
};

module.exports = { generateResponse };

File: full-stack_python/gemini_integration.py
# gemini_integration.py
import os
from google.generative_ai import GoogleGenerativeAI

api_key = "AIzaSyAi4RcPWCz6gP2dff23o65fXoi4a_9lXek" #os.getenv("GEMINI_API_KEY")
gen_ai = GoogleGenerativeAI(api_key)
model = gen_ai.get_model("gemini-1.5-flash")

def generate_response(prompt):
    try:
        result = model.generate_content(prompt)
        response_text = result['candidates'][0]['output'] if result['candidates'] else "No response generated."
        return response_text
    except Exception as e:
        print(f"Error generating content: {e}")
        return "Error generating content."
